### ISSUE TITLE
getSummonerId() on line 183 (php-riot-api.php)

### ISSUE THREAD

**Issue by Seismix** [kevinohashi/php-riot-api#43]
I ran into this problem while I was testing different names but I'm not sure if it's only me getting this.

Basically your function can't process names with spaces or UTF-8 chars (e.g. The Wild Goat or Ilùvátar or the like) especially because the `strtolower()` function doesn't work properly with UTF-8 chars. I managed to fix that by replacing `strtolower()` with `mb_strtolower()` and adding `str_replace()`:

```
public function getSummonerId($name) {
            $name = mb_strtolower(str_replace(" ", "",$name), 'UTF-8');
                        (...)
}
```

Maybe you can find a better solution or arrange it in a better way (I'm still a beginner).

Edit: I checked this https://github.com/kevinohashi/php-riot-api/issues/33 but I use double quotes and it doesn't work.

**Comment by kevinohashi** [comment]
That old solution definitely doesn't work anymore. Your proposed one doesn't seem to work for me either. I can't get it encoding properly.

I'm testing with 
$summoner_name = "Øhm";

https://na.api.pvp.net/api/lol/na/v1.4/summoner/by-name/Øhm?api_key=

That url works on testing from developer area, but with PHP I can't seem to get it to generate it properly right now. If you or anyone else has some thoughts on how to fix this, I'm open to ideas. I don't have that much time this second to really investigate it further.

**Comment by Seismix** [comment]
I tested with 
$summoner_name = "Øhm"
and it works.
```
//returns a summoner's id
	public function getSummonerId($name) {
			$name = mb_strtolower(str_replace(" ", "",$name), 'UTF-8');
			$summoner = $this->getSummonerByName($name);
			if (self::DECODE_ENABLED) {
				return $summoner[$name]['id'];
			}
			else {
				$summoner = json_decode($summoner, true);
				return $summoner[$name]['id'];
			}
	}	
```

-> tested with

```
$r = $test->getSummonerId('Øhm');
    print_r($r);
```
I get `38978760`

**Comment by kevinohashi** [comment]
Can you print how the url is encoded? I'm getting this:
https://na.api.pvp.net/api/lol/na/v1.4/summoner/by-name/%C3%98hm

Of course the API is down now :(

**Comment by Seismix** [comment]
I get
`../summoner/by-name/%C3%B8hm/..`

EUW is up btw.

**Comment by kevinohashi** [comment]
I'm getting service unavailable there too. 503.

**Comment by Seismix** [comment]
Yup API is down her as well.

**Comment by gregtampa** [comment]
so kevin so this working on NA now as is or do i need to modify the code to get it working with my example name? I'm so sorry I haven't kept up lately on the builds for your project.. I ended up rewriting the whole thing that would be more customized for what I needed the Riot API for.
I did some digging around and found out that these types of names can not be made anymore, as riot done away with using special chars. due to their new database infrastructure. Problem is old names still have them and still must account for them. The API is working now btw.
